itest/test/spectre/spectre_test.py

```diff
@@ -234,6 +240,106 @@ def test_post_request_not_cached(self):
+        response = post_through_spectre(
+            '/post_always_cache/',
+            data={},
+            extra_headers={'content-type': 'application/json'}
+        )
+        assert 200 == response.status_code
+        assert 'miss' == response.headers['Spectre-Cache-Status']
+
+        # When calling again the result should be cached
+        response = post_through_spectre(
+            '/post_always_cache/',
+            data={},
+            extra_headers={'content-type': 'application/json'}
+        )
+        assert 200 == response.status_code
+        assert 'hit' == response.headers['Spectre-Cache-Status']
+
+    def test_post_cache_hit_even_if_body_doesnt_match_without_vary(self):
+        response = post_through_spectre(
+            '/post_always_cache/',
+            data='{"field1":"key1"}',
+            extra_headers={'content-type': 'application/json'}
+        )
+        assert 200 == response.status_code
+        assert 'miss' == response.headers['Spectre-Cache-Status']
+
+        # When calling again the result should be cached
+        response = post_through_spectre(
+            '/post_always_cache/',
+            data='{"field1":"key2"}',
+            extra_headers={'content-type': 'application/json'}
+        )
+        assert 200 == response.status_code
+        assert 'hit' == response.headers['Spectre-Cache-Status']
+
+    def test_post_cached_with_id(self):
+        response = post_through_spectre(
+            '/post_id_cache/',
+            data='{"request_id":123, "vary_id":"abc"}',
+            extra_headers={'content-type': 'application/json'}
+        )
+        assert 200 == response.status_code
+        assert 'miss' == response.headers['Spectre-Cache-Status']
+
+        # When calling with different data, we will see a cache miss.
+        response = post_through_spectre(
+            '/post_id_cache/',
+            data='{"request_id":234, "vary_id":"abc"}',
+            extra_headers={'content-type': 'application/json'}
+        )
+        assert 200 == response.status_code
+        assert 'miss' == response.headers['Spectre-Cache-Status']
+
+        # When calling with different data with same id, we will see a cache miss.
+        response = post_through_spectre(
+            '/post_id_cache/',
+            data='{"request_id":234, "vary_id":"def"}',
+            extra_headers={'content-type': 'application/json'}
+        )
+        assert 200 == response.status_code
+        assert 'miss' == response.headers['Spectre-Cache-Status']
+
+        # Calling again with same request_id should be a cache hit
+        response = post_through_spectre(
+            '/post_id_cache/',
+            data='{"request_id":234, "vary_id":"abc"}',
+            extra_headers={'content-type': 'application/json'}
+        )
+        assert 200 == response.status_code
+        assert 'hit' == response.headers['Spectre-Cache-Status']
+
+    def test_post_cached_with_id_ignore_fields(self):
```

Comment:
Can you also test that when id extraction is enabled you can delete the cache for a specific id?

Reply:
Fixed